### ISSUE TITLE
Doom-emacs: wrong-type-argument stringp nil

### ISSUE THREAD

**Issue by lroolle** [cyrus-and/zoom#32]
I'm using doom-emacs, and config like this:

```lisp
(use-package zoom
  :after-call pre-command-hook
  :config
  (remove-hook 'window-configuration-change-hook #'zoom)
  (add-hook 'doom-switch-window-hook #'zoom)
  (custom-set-variables
   '(zoom-size '(0.7 . 0.7))
   '(zoom-ignored-major-modes '(dired-mode vterm-mode help-mode helpful-mode rxt-help-mode help-mode-menu org-mode))
   '(zoom-ignored-buffer-names '("*doom:scratch*" "*info*" "*helpful variable: argv*"))
   '(zoom-ignored-buffer-name-regexps '("^*calc" "\\*helpful variable: .*\\*"))
   '(zoom-ignore-predicates '((lambda () (> (count-lines (point-min) (point-max)) 20))))
   ))
```

This happens when open helpful-mode or org-mode edit-src

<img width="1486" alt="image" src="https://user-images.githubusercontent.com/14071051/87869492-e6294180-c9d2-11ea-9d58-ec6518c5749b.png">

<details>
<summary>Backtrace</summary>

```lisp
Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  doom-run-switch-buffer-hooks-a(#f(compiled-function (buffer-or-name &optional action frame) "Display BUFFER-OR-NAME in some window, without selecting it.\nBUFFER-OR-NAME must be a buffer or a string naming a live buffer.\nReturn the window chosen for displaying that buffer, or nil if no\nsuch window is found.\n\nOptional argument ACTION, if non-nil, should specify a buffer\ndisplay action of the form (FUNCTIONS . ALIST).  FUNCTIONS is\neither an \"action function\" or a possibly empty list of action\nfunctions.  ALIST is a possibly empty \"action alist\".\n\nAn action function is a function that accepts two arguments: the\nbuffer to display and an action alist.  Based on those arguments,\nit should try to display the buffer in a window and return that\nwindow.  An action alist is an association list mapping symbols\nto values.  Action functions use the action alist passed to them\nto fine-tune their behaviors.\n\n`display-buffer' builds a list of action functions and an action\nalist by combining any action functions and alists specified by\n`display-buffer-overriding-action', `display-buffer-alist', the\nACTION argument, `display-buffer-base-action', and\n`display-buffer-fallback-action' (in order).  Then it calls each\nfunction in the combined function list in turn, passing the\nbuffer as the first argument and the combined action alist as the\nsecond argument, until one of the functions returns non-nil.\n\nAction functions and the action they try to perform are:\n `display-buffer-same-window' -- Use the selected window.\n `display-buffer-reuse-window' -- Use a window already showing\n    the buffer.\n `display-buffer-in-previous-window' -- Use a window that did\n    show the buffer before.\n `display-buffer-use-some-window' -- Use some existing window.\n `display-buffer-pop-up-window' -- Pop up a new window.\n `display-buffer-below-selected' -- Use or pop up a window below\n    the selected one.\n `display-buffer-at-bottom' -- Use or pop up a window at the\n    bottom of the selected frame.\n `display-buffer-pop-up-frame' -- Show the buffer on a new frame.\n `display-buffer-in-child-frame' -- Show the buffer in a\n    child frame.\n `display-buffer-no-window' -- Do not display the buffer and\n    have `display-buffer' return nil immediately.\n\nAction alist entries are:\n `inhibit-same-window' -- A non-nil value prevents the same\n    window from being used for display.\n `inhibit-switch-frame' -- A non-nil value prevents any frame\n    used for showing the buffer from being raised or selected.\n `reusable-frames' -- The value specifies the set of frames to\n    search for a window that already displays the buffer.\n    Possible values are nil (the selected frame), t (any live\n    frame), visible (any visible frame), 0 (any visible or\n    iconified frame) or an existing live frame.\n `pop-up-frame-parameters' -- The value specifies an alist of\n    frame parameters to give a new frame, if one is created.\n `window-height' -- The value specifies the desired height of the\n    window chosen and is either an integer (the total height of\n    the window), a floating point number (the fraction of its\n    total height with respect to the total height of the frame's\n    root window) or a function to be called with one argument -\n    the chosen window.  The function is supposed to adjust the\n    height of the window; its return value is ignored.  Suitable\n    functions are `shrink-window-if-larger-than-buffer' and\n    `fit-window-to-buffer'.\n `window-width' -- The value specifies the desired width of the\n    window chosen and is either an integer (the total width of\n    the window), a floating point number (the fraction of its\n    total width with respect to the width of the frame's root\n    window) or a function to be called with one argument - the\n    chosen window.  The function is supposed to adjust the width\n    of the window; its return value is ignored.\n `preserve-size' -- The value should be either (t . nil) to\n    preserve the width of the chosen window, (nil . t) to\n    preserve its height or (t . t) to preserve its height and\n    width in future changes of the window configuration.\n `window-parameters' -- The value specifies an alist of window\n    parameters to give the chosen window.\n `allow-no-window' -- A non-nil value means that `display-buffer'\n    may not display the buffer and return nil immediately.\n\nThe entries `window-height', `window-width' and `preserve-size'\nare applied only when the window used for displaying the buffer\nnever showed another buffer before.\n\nThe ACTION argument can also have a non-nil and non-list value.\nThis means to display the buffer in a window other than the\nselected one, even if it is already displayed in the selected\nwindow.  If called interactively with a prefix argument, ACTION\nis t.  Non-interactive calls should always supply a list or nil.\n\nThe optional third argument FRAME, if non-nil, acts like a\n(reusable-frames . FRAME) entry appended to the action alist\nspecified by the ACTION argument." ... #<bytecode 0x44239489>) #<buffer *helpful variable: helm-ff-avfs-directory*> nil)
  apply(doom-run-switch-buffer-hooks-a #f(compiled-function (buffer-or-name &optional action frame) "Display BUFFER-OR-NAME in some window, without selecting it.\nBUFFER-OR-NAME must be a buffer or a string naming a live buffer.\nReturn the window chosen for displaying that buffer, or nil if no\nsuch window is found.\n\nOptional argument ACTION, if non-nil, should specify a buffer\ndisplay action of the form (FUNCTIONS . ALIST).  FUNCTIONS is\neither an \"action function\" or a possibly empty list of action\nfunctions.  ALIST is a possibly empty \"action alist\".\n\nAn action function is a function that accepts two arguments: the\nbuffer to display and an action alist.  Based on those arguments,\nit should try to display the buffer in a window and return that\nwindow.  An action alist is an association list mapping symbols\nto values.  Action functions use the action alist passed to them\nto fine-tune their behaviors.\n\n`display-buffer' builds a list of action functions and an action\nalist by combining any action functions and alists specified by\n`display-buffer-overriding-action', `display-buffer-alist', the\nACTION argument, `display-buffer-base-action', and\n`display-buffer-fallback-action' (in order).  Then it calls each\nfunction in the combined function list in turn, passing the\nbuffer as the first argument and the combined action alist as the\nsecond argument, until one of the functions returns non-nil.\n\nAction functions and the action they try to perform are:\n `display-buffer-same-window' -- Use the selected window.\n `display-buffer-reuse-window' -- Use a window already showing\n    the buffer.\n `display-buffer-in-previous-window' -- Use a window that did\n    show the buffer before.\n `display-buffer-use-some-window' -- Use some existing window.\n `display-buffer-pop-up-window' -- Pop up a new window.\n `display-buffer-below-selected' -- Use or pop up a window below\n    the selected one.\n `display-buffer-at-bottom' -- Use or pop up a window at the\n    bottom of the selected frame.\n `display-buffer-pop-up-frame' -- Show the buffer on a new frame.\n `display-buffer-in-child-frame' -- Show the buffer in a\n    child frame.\n `display-buffer-no-window' -- Do not display the buffer and\n    have `display-buffer' return nil immediately.\n\nAction alist entries are:\n `inhibit-same-window' -- A non-nil value prevents the same\n    window from being used for display.\n `inhibit-switch-frame' -- A non-nil value prevents any frame\n    used for showing the buffer from being raised or selected.\n `reusable-frames' -- The value specifies the set of frames to\n    search for a window that already displays the buffer.\n    Possible values are nil (the selected frame), t (any live\n    frame), visible (any visible frame), 0 (any visible or\n    iconified frame) or an existing live frame.\n `pop-up-frame-parameters' -- The value specifies an alist of\n    frame parameters to give a new frame, if one is created.\n `window-height' -- The value specifies the desired height of the\n    window chosen and is either an integer (the total height of\n    the window), a floating point number (the fraction of its\n    total height with respect to the total height of the frame's\n    root window) or a function to be called with one argument -\n    the chosen window.  The function is supposed to adjust the\n    height of the window; its return value is ignored.  Suitable\n    functions are `shrink-window-if-larger-than-buffer' and\n    `fit-window-to-buffer'.\n `window-width' -- The value specifies the desired width of the\n    window chosen and is either an integer (the total width of\n    the window), a floating point number (the fraction of its\n    total width with respect to the width of the frame's root\n    window) or a function to be called with one argument - the\n    chosen window.  The function is supposed to adjust the width\n    of the window; its return value is ignored.\n `preserve-size' -- The value should be either (t . nil) to\n    preserve the width of the chosen window, (nil . t) to\n    preserve its height or (t . t) to preserve its height and\n    width in future changes of the window configuration.\n `window-parameters' -- The value specifies an alist of window\n    parameters to give the chosen window.\n `allow-no-window' -- A non-nil value means that `display-buffer'\n    may not display the buffer and return nil immediately.\n\nThe entries `window-height', `window-width' and `preserve-size'\nare applied only when the window used for displaying the buffer\nnever showed another buffer before.\n\nThe ACTION argument can also have a non-nil and non-list value.\nThis means to display the buffer in a window other than the\nselected one, even if it is already displayed in the selected\nwindow.  If called interactively with a prefix argument, ACTION\nis t.  Non-interactive calls should always supply a list or nil.\n\nThe optional third argument FRAME, if non-nil, acts like a\n(reusable-frames . FRAME) entry appended to the action alist\nspecified by the ACTION argument." ... #<bytecode 0x44239489>) (#<buffer *helpful variable: helm-ff-avfs-directory*> nil))
  display-buffer(#<buffer *helpful variable: helm-ff-avfs-directory*> nil)
  #f(compiled-function (buffer-or-name &optional action norecord) "Display buffer specified by BUFFER-OR-NAME and select its window.\nBUFFER-OR-NAME may be a buffer, a string (a buffer name), or nil.\nIf it is a string not naming an existent buffer, create a buffer\nwith that name.  If BUFFER-OR-NAME is nil, choose some other\nbuffer.  In either case, make that buffer current and return it.\n\nThis uses `display-buffer' as a subroutine.  The optional ACTION\nargument is passed to `display-buffer' as its ACTION argument.\nSee `display-buffer' for more information.  ACTION is t if called\ninteractively with a prefix argument, which means to pop to a\nwindow other than the selected one even if the buffer is already\ndisplayed in the selected window.\n\nIf a suitable window is found, select that window.  If it is not\non the selected frame, raise that window's frame and give it\ninput focus.\n\nOptional third arg NORECORD non-nil means do not put this buffer\nat the front of the list of recently selected ones." (interactive #f(compiled-function () #<bytecode 0x1ffbdb1f1385>)) #<bytecode 0x43fd3f95>)(#<buffer *helpful variable: helm-ff-avfs-directory*>)
  apply(#f(compiled-function (buffer-or-name &optional action norecord) "Display buffer specified by BUFFER-OR-NAME and select its window.\nBUFFER-OR-NAME may be a buffer, a string (a buffer name), or nil.\nIf it is a string not naming an existent buffer, create a buffer\nwith that name.  If BUFFER-OR-NAME is nil, choose some other\nbuffer.  In either case, make that buffer current and return it.\n\nThis uses `display-buffer' as a subroutine.  The optional ACTION\nargument is passed to `display-buffer' as its ACTION argument.\nSee `display-buffer' for more information.  ACTION is t if called\ninteractively with a prefix argument, which means to pop to a\nwindow other than the selected one even if the buffer is already\ndisplayed in the selected window.\n\nIf a suitable window is found, select that window.  If it is not\non the selected frame, raise that window's frame and give it\ninput focus.\n\nOptional third arg NORECORD non-nil means do not put this buffer\nat the front of the list of recently selected ones." (interactive #f(compiled-function () #<bytecode 0x1ffbdb1a26c5>)) #<bytecode 0x43fd3f95>) #<buffer *helpful variable: helm-ff-avfs-directory*>)
  pop-to-buffer(#<buffer *helpful variable: helm-ff-avfs-directory*>)
  helpful-variable(helm-ff-avfs-directory)
  #f(compiled-function (x) #<bytecode 0x1ffbda7262d9>)("helm-ff-avfs-directory")
  ivy-call()
  ivy-read("Describe variable: " [:all-files hasNextPage ansi-color-apply-overlay-face epg-context-signers locals 0 -is-infix\? message-all-epg-keys-available-p magit-submodule-update edebug-trace epg--status-VALIDSIG magit-reflog-commit help-variable 0 org-emphasis-regexp-components mml-secure-smime-sign-with-sender url-cookie-multiple-line vc-hg--read-u8 no-activate nsm-permanent-host-settings doom-modeline-eldoc-bar 0 dired-at-point avl-tree--stack-store--cmacro cl-print--preprocess which-key-side-window-max-height selinux-context hide-mode-line-mode-map wicon content-type 0 0 cl--struct-epg-signature-p evil-visual-exchange-corners \" f-directory\? Edebug\ All\ Defs time-format Info-apropos-find-file \' :name-match helm-grep-mode-jump-other-window-forward tramp-cache-data org-babel-results-keyword gorepl-run-load-current-file magit-diff-wash-hunk not-at-beginning-of-line log-edit-set-header expired calendar-debug-sexp ...] :predicate counsel--variable-p :require-match t :history counsel-describe-symbol-history :keymap (keymap (67108908 . counsel--info-lookup-symbol) (67108910 . counsel-find-symbol)) :preselect "" :action #f(compiled-function (x) #<bytecode 0x1ffbda7262d9>) :caller counsel-describe-variable)
  counsel-describe-variable()
  funcall-interactively(counsel-describe-variable)
  call-interactively(counsel-describe-variable nil nil)
  command-execute(counsel-describe-variable)
```
</details>

<details>
<summary>Versions</summay>

- emacs: 
GNU Emacs 27.0.91
Copyright (C) 2020 Free Software Foundation, Inc.
GNU Emacs comes with ABSOLUTELY NO WARRANTY.
You may redistribute copies of GNU Emacs
under the terms of the GNU General Public License.
For more information about these matters, see the file named COPYING.

- doom-emacs: v2.0.9
</details>

**Comment by cyrus-and** [comment]
Hmm I don't see anything zoom-related in the backtrace, are you sure it is the culprit?

**Comment by cyrus-and** [comment]
> ```lisp
>   :config
>   (remove-hook 'window-configuration-change-hook #'zoom)
>   (add-hook 'doom-switch-window-hook #'zoom)
> ```

Moreover what's the point of this?

**Comment by lroolle** [comment]
> ```lisp
>   :config
>   (remove-hook 'window-configuration-change-hook #'zoom)
>   (add-hook 'doom-switch-window-hook #'zoom)
> ```
> 
> 
> 
> Moreover what's the point of this?

It's doom's hooks  according to this  issue: @hlissner
-> https://github.com/hlissner/doom-emacs/issues/2225#issuecomment-568287835

Well actually after I remove these two hooks the Errors never occur. But zoom won't' work automatically any more. 

Thanks for replies and due to this is doom's Issue, I'll close this issue.

**Comment by cyrus-and** [comment]
> But zoom won't' work automatically any more.

Try with:

```lisp
(custom-set-variables
  '(zoom-mode t))
```

**Comment by lroolle** [comment]
> > But zoom won't' work automatically any more.
> 
> Try with:
> 
> ```lisp
> (custom-set-variables
>   '(zoom-mode t))
> ```

While seems like doom does  not respect `zoom-mode` on...this doesnt work automatically

**Comment by cyrus-and** [comment]
Hmm, I'm not familiar with Doom but that's enough to make it work automatically with *regular* Emacs, being `zoom-mode` a global minor mode.

**Comment by hlissner** [comment]
@troole This works for me:

```elisp
(use-package zoom
  :hook doom-switch-window
  :config
  (setq zoom-size '(0.7 . 0.7)
        zoom-ignored-major-modes '(dired-mode vterm-mode help-mode helpful-mode rxt-help-mode help-mode-menu org-mode)
        zoom-ignored-buffer-names '("*doom:scratch*" "*info*" "*helpful variable: argv*")
        zoom-ignored-buffer-name-regexps '("^\\*calc" "\\*helpful variable: .*\\*")
        zoom-ignore-predicates (list (lambda () (> (count-lines (point-min) (point-max)) 20)))))
```

Although, that ignore predicate is pretty aggressive. I'll assume that was intended.

**Comment by lroolle** [comment]
@hlissner Awesome! 

Thanks for the following.

**Comment by hlissner** [comment]
@lroolle After some investigation, turns out this wasn't actually a bug, it's just necessary to enable `zoom-mode` in order for it to advise `pre-redisplay-function` -- where it calls `zoom--update` from a stable "current window". 

Without this, it can lose track of the current window where complex `display-buffer-alist` rules are involved. e.g. with Doom's popup manager, where `display-buffer-in-side-window` is used, _which_ will delete existing windows to redisplay them on the side of the frame. In that case, `zoom--update` operates out of a delete window, causing the error.

TL;DR, this should fix it:

```elisp
(use-package zoom
  :hook (doom-first-input . zoom-mode)
  :config
  (setq zoom-size '(0.7 . 0.7)
        zoom-ignored-major-modes '(dired-mode vterm-mode help-mode helpful-mode rxt-help-mode help-mode-menu org-mode)
        zoom-ignored-buffer-names '("*doom:scratch*" "*info*" "*helpful variable: argv*")
        zoom-ignored-buffer-name-regexps '("^\\*calc" "\\*helpful variable: .*\\*")
        zoom-ignore-predicates (list (lambda () (> (count-lines (point-min) (point-max)) 20)))))
```

**Comment by cyrus-and** [comment]
Thank you @hlissner for taking the time to dig into this.
